### PR TITLE
[Metis] rename metis_application to MetisApplication

### PR DESCRIPTION
**Pull request by philbucher**
I searched, I think I found all places where renaming was necessary

Users should not be affected since we always import `KratosMultiphysics.MetisApplication`

### CODE REVIEW

File: applications/trilinos_application/custom_utilities/trilinos_cutting_app.h [hunk 1]
```diff
@@ -14,7 +14,7 @@
 #include "utilities/math_utils.h"
 #include "geometries/triangle_3d_3.h"
 #include "custom_utilities/parallel_fill_communicator.h"
-#include "metis_application/custom_processes/set_mpi_communicator_process.h"
+#include "MetisApplication/custom_processes/set_mpi_communicator_process.h"
```
Comment (philbucher) [inline applications/trilinos_application/custom_utilities/trilinos_cutting_app.h]: @jcotela for future reference, we should refactor this dependency

**Comment by RiccardoRossi** [comment]
@philbucher would you agree on having this at the beginning of the next release cycle? i am telling since i also think we should invert dependency of Trilinos on application to a dependency of other applications from trilinos, but of course this is something to be discussed very carefully

**Comment by philbucher** [comment]
> @philbucher would you agree on having this at the beginning of the next release cycle? i am telling since i also think we should invert dependency of Trilinos on application to a dependency of other applications from trilinos, but of course this is something to be discussed very carefully

@RiccardoRossi yes we can do that.
Inverting the dependencies should be done in a separate PR (preferably following this one, then the files won't have to be renamed)

File: applications/trilinos_application/custom_utilities/trilinos_cutting_app.h (hunk 1 above)
Comment (philbucher) [reply applications/trilinos_application/custom_utilities/trilinos_cutting_app.h]: done in #4057

**Comment by philbucher** [comment]
alright, release is done, conflicts are solved, this is ready for being merged :)

**Comment by philbucher** [comment]
ping

**Comment by roigcarlo** [comment]
Same as in #3772, we should look if the history lost is only on github or if it is a git problem.

**Comment by philbucher** [comment]
True, no clue why, I thought I had done it right.
I will check 
Thx for noticing

**Comment by philbucher** [comment]
see answer in the other PR #3772

**Review by jcotela — APPROVED** [review]
@KratosMultiphysics/technical-committee approves

**Review by jcotela — DISMISSED** [review]
@KratosMultiphysics/technical-committee approves.

**Comment by philbucher** [comment]
I fixed a conflict

**Review by jcotela — APPROVED** [review]
Re-approving

**Pull request merged**